presto-main/src/main/java/com/facebook/presto/sql/planner/plan/StatisticAggregations.java

```diff
@@ -25,31 +25,51 @@
 import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableMap;
 
+import javax.annotation.concurrent.Immutable;
+
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.stream.Collectors;
 
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class StatisticAggregations
 {
+    // Added to preserve the order for serializing and deserializing
+    // map in c++ worker doesn't represent insertion order
+    private final List<VariableAggregation> aggregationList;
     private final Map<VariableReferenceExpression, Aggregation> aggregations;
     private final List<VariableReferenceExpression> groupingVariables;
 
     @JsonCreator
     public StatisticAggregations(
-            @JsonProperty("aggregations") Map<VariableReferenceExpression, Aggregation> aggregations,
+            @JsonProperty("aggregationList") List<VariableAggregation> aggregationList,
```

Comment:
aggregationList -> aggregations for consistency

Note that groupingVariables is also a List, but doesn't have "List" suffix.